spec/create-resource.md

```diff
@@ -0,0 +1,35 @@
+
+* **Media type:** The MIME type or format of the content (e.g., `text/plain, application/json`). This helps the server understand how to store and serve the content. It may be mandatory for some protocols when content is provided (and may be omitted if creating an empty container with no content).
+
+**Behavior:**
+
+* If a *target container* is provided, the server will create a new resource as a member of that container. The server determines the new resource’s identifier (often incorporating the suggested name if provided and allowed). The new resource is then added to the specified container’s membership.
+
+* If a *full target identifier* (including the desired name/path) is provided, the server will attempt to create a resource at that exact identifier. If a full target identifier is provided and a resource already exists at that identifier, the server **MUST** fail with a Conflict response, unless the client explicitly indicates that overwriting is acceptable.
+
+* The server **MAY** enforce additional constraints on creation. For example, it may have size limits, quota checks, or restrictions on allowed media types or file extensions. If any such constraint is violated (e.g., the content is too large or an unsupported type), the server will reject the operation with an appropriate error.
+
+* On successful creation, the new resource is persistently stored and linked into the logical container hierarchy. The server will assign an identifier if needed and make the resource available for future operations. The identifier (e.g., a URI or path) of the newly created resource **MUST** be returned to the client (so the client knows how to refer to it). The creation is atomic – if any step fails, no new resource is created (or any partial resource is cleaned up).
+
+**Possible Responses:** *(on the abstract operation level, not tied to a specific protocol)*
+
+* **Created:** The operation succeeded in creating a new resource. The response includes the new resource’s identifier and possibly a minimal representation or metadata (e.g., an ETag or version number).
+
+* **Bad Request:** The request was malformed or violated constraints (e.g., missing required inputs, invalid identifier, disallowed media type, or size limits exceeded). The resource was not created.
+
+* **Not Permitted:** The client is not authorized to create a resource in the target location.
+
+* **Conflict:** A resource at the target identifier already exists and the server does not allow overwrite (or a naming conflict occurred). No new resource was created.
+
+* **Not Found:** (If applicable) The specified target container does not exist or cannot be found. This could also be treated as a type of Bad Request or Not Permitted depending on the cause (e.g., if the container is missing vs. the client lacking access).
```

Comment:
```suggestion
* **Not Found:** (If applicable) The specified target container does not exist or cannot be found. This could also be treated as a type of Bad Request or Not Permitted, depending on the cause (e.g., if the container is missing or the client lacks access permission).
```